packages/vscode-extension/src/commonlib/m365Login.ts

```diff
@@ -121,6 +123,10 @@ export class M365Login extends BasicLogin implements M365TokenProvider {
     isInitiatedFromTdp?: boolean
   ): Promise<Result<string, FxError>> {
     if (tokenRequest.showDialog === undefined || tokenRequest.showDialog) {
+      const vscodeEnv = detectVsCodeEnv();
```

Comment:
Move this logic to constructor? I believe the vscode env does not change so just set `isCodeSpace` on initializing.
In addition, `updateIsCodeSpace` is not needed. Just set `isCodeSpace` when creating codeFlowLogin.

Reply:
Detect `codespace` using environment variable.